x-pack/plugin/esql/src/test/java/org/elasticsearch/xpack/esql/expression/function/scalar/math/BucketTests.java

```diff
@@ -131,6 +171,34 @@ private static TestCaseSupplier.TypedData numericBound(String name, DataType typ
         return new TestCaseSupplier.TypedData(v, type, name).forceLiteral();
     }
 
+    private static void numberCasesWithSpan(List<TestCaseSupplier> suppliers, String name, DataType numberType, Supplier<Number> number) {
+        suppliers.add(new TestCaseSupplier(name, List.of(numberType, DataTypes.DOUBLE), () -> {
+            List<TestCaseSupplier.TypedData> args = new ArrayList<>();
+            args.add(new TestCaseSupplier.TypedData(number.get(), "field"));
+            args.add(new TestCaseSupplier.TypedData(50., DataTypes.DOUBLE, "span").forceLiteral());
```

Comment:
Maybe we could randomize the span here.

Reply:
It could be randomised, though the matcher (and the style of these tests) would then need to be updated. I've kept it as is for now, but can be convinced of otherwise. :)

Reply:
Probably okay either way, I'll leave that to your discretion :)